subsys/bluetooth/mesh/beacon.c

```diff
@@ -529,7 +529,6 @@ static bool secure_beacon_authenticate(struct bt_mesh_subnet *sub, void *cb_data
 	return false;
 }
 
-#if defined(CONFIG_BT_MESH_V1d1)
 static bool priv_beacon_decrypt(struct bt_mesh_subnet *sub, void *cb_data)
```

Comment:
Now in theory, it looks like Mesh 1.0 node will be able to receive and decrypt private beacon, this seems at odds with the commit message? May be for that reason we may need BT_MESH_V101. Alternatively, we could also wrap this up behind SHA256 algorithm option, which seems like a proxy for Mesh 1.1 support?

Reply:
Good point. Using SHA256 option to enable mesh-1.1 won't be that intuitive. So perhaps we should leave 1.1 option and only remove legacy transport?

Reply:
Didn't it remove support for v1.0.1 in general? It claims backward compatibility with v1.0.1 but it is not v1.0.1 anymore.
After removing this option it is not possible to build mesh v1.0.1. It is possible to build 1.1 that passes PTS 1.0.1 tests.

Reply:
I agree with Alex, the idea is that "Mesh 1.0 node" is not something you can compile after this PR goes in. Only "Mesh 1.1 node". Which according to spec is backward compatible so it can be dropped into any network.

Reply:
Discussed offline: I'll rephrase the PR description and comment to clearly state that this PR makes the stack in Zephyr mesh 1.1 compatible, not mesh 1.0.1. To make it back mesh 1.0.1 compatible, the private beacons reception and SHA256 algorithm should be compiled out.